Publish the development branch to master

AlessandroA opened the pull request:
This PR is to publish the changes developed in the `dev` branch into `master`.

Changelog:

* New APIs to disable/re-enable all IRQs.
* New abstraction layer for context switches.
* Export tables.
* Hooks for privileged IRQ handlers.
* Secure boxes initialization.
* New page allocator feature.
* New support for uVisor-aware dynamic memory allocation.
* Keil RTX support on mbed OS.
* Removal of the secure gateway feature.
* Register gateway implemented (and APIs updated).
* New RPC APIs (header files only, for now).
* Many style improvements.
* Updated docs.
* Bugfixes.

@meriac @Patater @niklas-arm

Patater commented:
Remove the reverted commits and the reverting commits.
Remove training whitespace from `Docs: Add linker script padding for ARMv7-M MPUs`.

Then LGTM. Thanks.

AlessandroA commented:
* Removed 2 commits and their subsequent revert commits (total of 4 commits removed).
* Removed trailing whitespaces from the commit `Docs: Add linker script padding for ARMv7-M MPUs`.

Ready for merge.
